examples/notebooks/beam-ml/rag_usecase/redis_connector.py

```diff
@@ -0,0 +1,362 @@
+
+"""This module implements IO classes to read write documents in Redis.
+
+
+Insert Doc in Redis:
+-----------------
+:class:`InsertDocInRedis` is a ``PTransform`` that writes key and values to a
+configured sink, and the write is conducted through a redis pipeline.
+
+The ptransform works by getting the first and second elements from the input,
+this means that inputs like `[k,v]` or `(k,v)` are valid.
+
+Example usage::
+
+  pipeline | InsertDocInRedis(host='localhost',
+                          port=6379,
+                          batch_size=100)
+"""
+
+
+class InsertDocInRedis(PTransform):
+    """InsertDocInRedis is a ``PTransform`` that writes a ``PCollection`` of
+    key, value tuple or 2-element array into a redis server.
+    """
+
+    def __init__(self,
+                 host: str,
+                 port: int,
+                 command: Optional[str] = None,
+                 batch_size: int = 100
+                 ):
+
+        """
+
+        Args:
+        host (str): The redis host
+        port (int): The redis port
+        batch_size(int): Number of key, values pairs to write at once
+        command (str): command to be executed with redis client
+
+        Returns:
+        :class:`~apache_beam.transforms.ptransform.PTransform`
+
+        """
+
+        self._host = host
+        self._port = port
+        self._command = command
+        self._batch_size = batch_size
+
+    def expand(self, pcoll):
+        return pcoll \
+               | "Reshuffle for Redis Insert" >> Reshuffle() \
+               | "Insert document into Redis" >> beam.ParDo(_InsertDocRedisFn(self._host,
+                                                                              self._port,
+                                                                              self._command,
+                                                                              self._batch_size)
+                                                            # .with_output_types(Document)
+                                                            )
+
+
+class _InsertDocRedisFn(DoFn):
+    """Abstract class that takes in redis
+    credentials to connect to redis DB
+    """
+
+    def __init__(self,
+                 host: str,
+                 port: int,
+                 command: Optional[str] = None,
+                 batch_size: int = 100
+                 ):
+        self.host = host
+        self.port = port
+        self.command = command
+        self.batch_size = batch_size
+
+        self.batch_counter = 0
+        self.batch = list()
+
+        self.text_col = None
+
+    def finish_bundle(self):
+        self._flush()
+
+    def process(self, element, *args, **kwargs):
+        # if type(element) is not dict:
+        #     element = element._asdict()
+        self.batch.append(element)
+        self.batch_counter += 1
+        if self.batch_counter >= self.batch_size:
+            self._flush()
+            # yield Document(**element)
```

Comment:
Same thing here, if the commented code isn't going to be used now it's better to get rid of it and replace it with a TODO for future improvment